scripts/migrations/write_prefs_to_store.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env python3
+"""
+Copies all patrons' preferences to the `store` tables.
+"""
+import argparse
+from pathlib import Path
+
+from psycopg2 import DatabaseError
+
+import infogami
+from openlibrary.accounts import RunAs
+from openlibrary.accounts.model import OpenLibraryAccount
+from openlibrary.config import load_config
+from openlibrary.core import db
+from scripts.utils.graceful_shutdown import init_signal_handler, was_shutdown_requested
+
+DEFAULT_CONFIG_PATH = "/olsystem/etc/openlibrary.yml"
+PREFERENCE_TYPE = 'preferences'
+
+
+def setup(config_path):
+    init_signal_handler()
+    if not Path(config_path).exists():
+        raise FileNotFoundError(f'no config file at {config_path}')
+    load_config(config_path)
+    infogami._setup()
+
+
+def copy_preferences_to_store(keys, verbose: bool = False) -> list[str]:
+    errors = []
+    for key in keys:
+        if was_shutdown_requested():
+            break
+        try:
+            if verbose:
+                print(f"Writing {key} to store...")
+            username = key.split('/')[-1]
+            ol_acct = OpenLibraryAccount.get_by_username(username)
+            prefs = (ol_acct and ol_acct.get_user().preferences()) or {}
+            if ol_acct and prefs.get('type', '') != PREFERENCE_TYPE:
+                prefs['type'] = PREFERENCE_TYPE
+                prefs['_rev'] = None
+
+                with RunAs(username):
+                    ol_acct.get_user().save_preferences(
+                        prefs, msg="Update preferences for store", use_store=True
+                    )
```

Comment:
I wonder if this should be run as an administrator, as the patron will not be triggering this `save_preferences` call themself.  I think that we have an `as_admin` function that can help.